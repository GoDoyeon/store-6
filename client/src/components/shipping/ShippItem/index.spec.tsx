import { render } from '@testing-library/react';
import 'jest-styled-components';

import ShippingItem from './index';

describe('<ShippingItem />', () => {
  it('should render same with snapshot', () => {
    const changeSelectedBtn = () => {};
    const removeShippingItem = () => {};
    const modifyBtnClick = () => {};
    const selected = true;
    const shipItem = {
      idx: 1,
      name: 'test',
      phone: 'test',
      code: 'test',
      address: 'test',
      detailAddress: 'test',
      selected: 1,
    };

    const { container } = render(
      <ShippingItem
        changeSelectedBtn={changeSelectedBtn}
        removeShippingItem={removeShippingItem}
        modifyBtnClick={modifyBtnClick}
        selected={selected}
        shipItem={shipItem}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
