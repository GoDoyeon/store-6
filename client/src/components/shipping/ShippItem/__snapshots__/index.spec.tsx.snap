// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShippingItem /> should render same with snapshot 1`] = `
.c2 {
  position: relative;
  height: 16px;
}

.c4 {
  height: 100%;
}

.c3 {
  position: absolute;
  bottom: 5px;
  left: 1px;
}

.c0 {
  padding: 41px 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 15px;
}

.c0 > * {
  margin-left: 70px;
}

.c1 {
  width: 20px;
  cursor: pointer;
}

.c5 {
  line-height: 1.4;
  width: 150px;
}

.c5 div:last-child {
  color: #999999;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 370px;
  line-height: 1.4;
  white-space: pre-wrap;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 10px;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
  gap: 15px;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 7px;
}

.c8 img:last-child {
  opacity: 0;
}

.c8:hover img {
  opacity: 1;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <img
          class="c3"
        />
        <img
          class="c4"
        />
      </div>
    </div>
    <div
      class="c5"
    >
      <div>
        test
      </div>
      <div>
        test
      </div>
    </div>
    <div
      class="c6"
    >
      <div>
        test
      </div>
      <div>
        test
      </div>
    </div>
    <div
      class="c7"
    >
      <div
        class="c8"
      >
        <div
          class="c9"
        >
          <img
            alt="modify"
          />
          <div>
            수정
          </div>
        </div>
        <img
          alt="underline"
        />
      </div>
      <div
        class="c8"
      >
        <div
          class="c9"
        >
          <img
            alt="cancle"
          />
          <div>
            삭제
          </div>
        </div>
        <img
          alt="underline"
        />
      </div>
    </div>
  </div>
</div>
`;
